cli/Valet/Site.php

```diff
@@ -145,6 +145,71 @@ function parked()
         return $parkedLinks;
     }
 
+    /**
+     * Get all sites which are proxies (not Links, and contain proxy_pass directive)
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    function proxies()
+    {
+        $dir = VALET_HOME_PATH.'/Nginx/';
+        $tld = $this->config->read()['tld'];
+        $links = $this->links();
+        $parked = $this->parked();
+        $certsPath = VALET_HOME_PATH.'/Certificates';
+        $certs = $this->getCertificates($certsPath);
+
+        $proxies = collect($this->files->scandir($dir))
+        ->filter(function ($site, $key) use ($tld) {
+            // keep sites that match our TLD
+            return ends_with($site, '.'.$tld);
+        })->map(function ($site, $key) use ($tld) {
+            // remove the TLD suffix for consistency
+            return str_replace('.'.$tld, '', $site);
+        })->reject(function ($site, $key) use ($links) {
+            return $links->has($site);
+        })->reject(function ($site, $key) use ($parked) {
+            return $parked->has($site);
```

Comment:
If I understand this correctly, this means in theory we are not expecting sites that are otherwise parked to be able to be proxied?

Here is my situation:

 * All projects live in ~/Code
 * I have a Docker-based project in ~/Code/app.example.com
 * I expected I'd be able to `valet proxy app.example.com https://127.0.0.1:8443` to allow me to add app.example.com.test

This **actually works**, currently. However, afterwards, when I run `valet proxies` it does not show up in the list.

I see two potential fixes:

 * Remove this check here so that "parked proxies" would show up correctly in the `valet proxies` list (my favorite solution!)
 * Keep this check but block the addition of a proxy if the site is otherwise parked (not my favorite but would understand if that is the way it needs to be)

Reply:
Oh! Actually, I just did an experiment. If i removed the `parked` check things got super wonky. LOL. So, I removed it and added another check after the map to remove anything that says `(other)`. Kinda janky, and maybe there are better solutions, but trying these changes locally, this seems to work the best for me right now.

```
...
        $proxies = collect($this->files->scandir($dir))
        ->filter(function ($site, $key) use ($tld) {
            // keep sites that match our TLD
            return ends_with($site, '.'.$tld);
        })->map(function ($site, $key) use ($tld) {
            // remove the TLD suffix for consistency
            return str_replace('.'.$tld, '', $site);
        })->reject(function ($site, $key) use ($links) {
            return $links->has($site);
        })->mapWithKeys(function ($site) {
            $host = $this->getProxyHostForSite($site) ?: '(other)';
            return [$site => $host];
        })->reject(function ($host, $site) {
            return $host === '(other)';
        })->map(function ($host, $site) use ($certs) {
            $secured = $certs->has($site);
            $url = ($secured ? 'https': 'http').'://'.$site;
...
```

Maybe this is too heavy-handed and removes some of the flexibility of the "show proxies"? I don't know. It *should* list all of the proxies that are managed by Valet directly, though, so might be a solid enough solution for now.

Reply:
I think it makes sense. I've added it with some explanatory comments.